### ISSUE TITLE
Wider video size: is it possible?

### ISSUE THREAD

**Issue by biva** [nextcloud/spreed#1806]
Currently, the size of the video is 4:3 (or something similar). It's almost square, and not wide. With Skype, the video is wider: we can see more people in the room. I think it's something like 16:9 or 16:10

Is it possible to make the video wider?

Thanks!
(I'm using Firefox)

It might be related to the video resolution https://github.com/nextcloud/spreed/issues/504

**Comment by biva** [comment]
Maybe some insights here: https://github.com/webrtcHacks/WebRTC-Camera-Resolution

**Comment by biva** [comment]
Hello, any update regarding this enhancement request? Thanks! :)

**Comment by nickvergessen** [comment]
Not really, it is also only in our backlog. So we are not working actively on the issue.

**Comment by biva** [comment]
@nickvergessen I found different sites where they can choose the resolution:
- https://webrtchacks.github.io/WebRTC-Camera-Resolution/
- https://blog.addpipe.com/getusermedia-video-constraints/
- https://www.html5rocks.com/en/tutorials/getusermedia/intro/
- https://w3c.github.io/mediacapture-main/getusermedia.html#media-track-constraints

Do you think it would bring a solution? This issue is preventing us to use Talk with larger groups because we can't see everybody. Thanks :)

**Comment by nickvergessen** [comment]
Well good news, the way big calls are working will change with #1056 anyway.

**Comment by biva** [comment]
Ho, @nickvergessen there might be a misunderstanding. My point in larger groups is when we have several people in the same room, in front of the same camera: the video display has to be wide enough to see everybody in the room: I can't see everybody with a 4:3 format. Is it clearer?

**Comment by biva** [comment]
It looks like Constraint should be used, maybe this article gives some useful information (I'm not a developer, sorry if it doesn't help): https://webrtc.org/getting-started/media-capture-and-constraints

I also mentioned several websites using different resolutions above https://github.com/nextcloud/spreed/issues/1806#issuecomment-555623607

**Comment by ruedigerkupper** [comment]
See also #3407.

**Comment by ruedigerkupper** [comment]
I tried Jitsi today, which is WebRTC too, same browser, same camera. It instantly accessed my camera in 16:9 HD. In contrast, Nextcloud Talk accesses it 3:4 low resolution. So it seems there must be a way of selecting the mode?

**Comment by biva** [comment]
Hello @nickvergessen , would you know if there's any news on this topic?

**Comment by hoergen** [comment]
Any news? Would like to use it mainly with desktop sharing, nut the fixed 4:3 aspect ratio is a show stopper.

**Comment by ChildLearningClub** [comment]
maybe we will get it for Christmas :)

**Comment by pmlejnek** [comment]
I really need this feature. I use 16:9 aspect ratio and I use my own blackboard through OBS. Talk always trims my video on both sides (left and right) to 4:3. So I cannot share anything this way.
The strange thing is that if I use Firefox this doesn't happen, but there is another issue (the video becomes laggy and slow) on other participants' screens. With Chrome this doesn't happen, but I have the mentioned problem. So I cannot use one  or another.

**Comment by lfirewall1243** [comment]
Anything new to it?
Would like to use nextcloud talk instead Jitsi, but with the bad and cropped video it’s not really usable.

**Comment by llamallama** [comment]
This is still a problem

**Comment by DaRacci** [comment]
This is fairly important issue I believe is this on the roadmap at all?

**Comment by Denkkracht** [comment]
It is about showing 4:3 in 16:9 mode.

Zoom has a button for it in the Video settings tab: HD video.
Maybe that would be a solution too for Nextcloud Talk: to force a signal to 16:9. 

Anyway, I found a detour: in OBS you can set the 4:3 input to 16:9. And start Virtual camera. In Talk you choose OBS as source (OBS Virtual camera). 
But it is a bit unstable.
